dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -125,7 +134,23 @@ private static DependencyNode resolveCompileTimeDependencies(List<Artifact> depe
     }
     return result;
   }
-  
+
+  /**
+   * Finds the full dependency graph including duplicates and
+   * @param artifact
+   * @return
+   * @throws DependencyCollectionException
+   * @throws DependencyResolutionException
+   */
+  public static DependencyGraph getStaticLinkageCheckDependencies(Artifact artifact)
+      throws DependencyCollectionException, DependencyResolutionException {
+    DependencyNode node = resolveCompileTimeDependencies(artifact, true);
+    DependencyGraph graph = new DependencyGraph();
+    levelOrder(node, graph, true, true);
```

Comment:
At the call site, it's hard to figure out what boolean arguments are for, especially when there are multiple. I would consider using enums instead so that it is self-descriptive.

Reply:
Good point. Made an enum of

```
  private enum LevelOrderOption { NONE, FULL_DEPENDENCY, FULL_DEPENDENCY_WITH_PROVIDED }
```